Add detekt icon

hbmartin opened the pull request:
![detekt](https://github.com/user-attachments/assets/dcb18b19-477a-4975-92ff-cefcc45cbee8)


**Popularity metric:**

Detekt is the primary static analysis tool for the Kotlin programming language (Android).
It has [over 6k GH](https://github.com/detekt/detekt) stars and [outranks plugins](https://mvnrepository.com/open-source/gradle-plugins?p=2) from e.g. Google and Netflix.

<img width="875" alt="Screenshot 2024-09-03 at 4 20 50 PM" src="https://github.com/user-attachments/assets/d0b26826-efd5-472a-a85a-1ce2dd250d5a">


### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

- SVG was downloaded from detekt.dev and edited with Sketch
- Primary color was selected from reddish gradient color (red is typically associated with chekcers, the purple color is associated with Kotlin proper)

hbmartin commented:
@uncenter @LitoMore could you take a look at this? 🙏

uncenter commented:
> I'm struggling to get the SVG to exactly 24 width - I've tried a couple of approaches and always seem to be off by 0.001

Hmm maybe try optimizing with a higher precision? 4 should be good.

uncenter commented on icons/detekt.svg:
Hmm I'm seeing differences @LitoMore:

![CleanShot 2024-09-27 at 13 04 43](https://github.com/user-attachments/assets/8cadd6db-43dd-4dfe-a135-57c069a9996b)

Same result with both https://detekt.dev/img/favicon.svg and https://detekt.dev/img/home/detekt-logo.svg from source header.

uncenter replied:
I believe this has yet to be resolved?

uncenter commented on _data/simple-icons.json:
```diff
@@ -4162,6 +4162,11 @@
+        {
+            "title": "Detekt",
+            "hex": "45413C",
+            "source": "https://detekt.dev"
```
```suggestion
            "source": "https://github.com/detekt/detekt/blob/bccfd399868a06924fbae3a310a5af7082260ffb/website/static/img/favicon.svg"
```

uncenter commented on _data/simple-icons.json:
```diff
@@ -4162,6 +4162,11 @@
         },
+        {
+            "title": "Detekt",
+            "hex": "45413C",
```
As per https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#color-guidelines, we *usually* use https://meyerweb.com/eric/tools/color-blend/ for logos with gradients like this. I inputted the two start/stop colors from the gradient (https://meyerweb.com/eric/tools/color-blend/#7942EE:FF4651:1:hex) and got `#BC44A0` as the middle value. However, if you believe there is a good reason to pick one of the gradient colors over the other instead of the middle/mixed color, I'll hear you out :D

```suggestion
            "hex": "BC44A0",
```

hbmartin replied:
I felt that the reddish color was more meaningful given that detekt is a quality checking tool and reds are generally associated with the errors / complaints that the tool emits. The purplish midpoint seemed less semantically relevant and doesn't really connote anything about detekt in my mind.
Happy either way ultimately 🤷

uncenter replied:
Actually, let's do neither the midpoint or red... let's do the purple!
![detekt dev_docs_gettingstarted_gradle](https://github.com/user-attachments/assets/d1d84f2d-2f8a-4fc5-afca-9154e4310d0d)

They seem to use it as their accent color on the docs, seems fitting. Does that work for you?

uncenter replied:
Edited the suggestion above to use `#7942EE` 🫡

hbmartin replied:
@uncenter excellent! 🙏

uncenter commented on _data/simple-icons.json:
```diff
@@ -4162,6 +4162,11 @@
             "source": "https://www.spiegel.de"
         },
+        {
+            "title": "Detekt",
```
```suggestion
            "title": "detekt",
```

This is how they capitalize it.

LitoMore reviewed (changes requested):
![](https://github.com/user-attachments/assets/cffddcb9-5e8a-4b91-84b6-25a0b24bb307)

Differences need to be resolved.

hbmartin commented:
@LitoMore @uncenter Thanks for catching the diff, I've updated SVG path and PR description.
What tool(s) are you using to make the comparisons? (I'd like to be able to  better check myself.)

uncenter commented:
> @LitoMore @uncenter Thanks for catching the diff, I've updated SVG path and PR description. What tool(s) are you using to make the comparisons? (I'd like to be able to better check myself.)

I just use Inkscape; I open the contributed icon version, import/add the original source file, resize to match, and compare the visual difference.

PeterShaggyNoble reviewed (changes requested):
SVG looks good to me now, @hbmartin 👍🏻 Although, there are a couple of excess points in the magnifying glass that could be removed, and possibly in the exclamation points as well.

For the colour, I'd suggest `#8943dc` as it's assign to the variable ` --ifm-color-primary` in their website's stylesheet and, with that, the `source` URL would need to be updated to https://detekt.dev/

hbmartin commented:
Thanks for those points @PeterShaggyNoble - I've simplified those circular paths, did an "onion skin" check in inkscape with the original logo, and updated the color 👍

hbmartin commented on _data/simple-icons.json:
```diff
@@ -13295,20 +13300,20 @@
 		"aliases": {
 			"dup": [
 				{
-					"title": "France Inter",
```
These changes appear to be coming from the commit hook, I've tried to undo them but perhaps this is correct?

PeterShaggyNoble reviewed (changes requested):
Thanks, @hbmartin. However, the checks are now [failing](https://github.com/simple-icons/simple-icons/actions/runs/15046384633/job/42571625287?pr=11723) as the precision was set too high when optimising - you used 6, while the maximum we allow is 5.

hbmartin commented:
@PeterShaggyNoble *face palm* ... fixed the precision issue, lint and test are passing for me locally now

hbmartin commented:
@PeterShaggyNoble tbh I don't have any more time to spend on this, so feel free to close this PR.
